app/bundles/EmailBundle/Model/EmailModel.php

```diff
@@ -1023,12 +1023,17 @@ public function sendEmail($email, $leads, $options = [])
                 $dnc = $emailRepo->getDoNotEmailList();
             }
 
-            //weed out do not contacts
+
             if (!empty($dnc)) {
                 foreach ($sendTo as $k => $lead) {
+                    //weed out do not contacts
                     if (in_array(strtolower($lead['email']), $dnc)) {
                         unset($sendTo[$k]);
                     }
+                    //apply frequency rules
+                    if($this->mailHelper->applyFrequencyRules($lead)){
```

Comment:
`$lead` here is an array not a `Lead` entity which should fail for applyFrequencyRules()` since it's type hinted for `Lead`.